src/Nethermind/Nethermind.JsonRpc/JsonRpcProcessor.cs

```diff
@@ -67,8 +75,45 @@ private void BuildTraceJsonSerializer()
             _traceSerializer = JsonSerializer.Create(jsonSettings);
         }
 
+        string filePath;
+        int fileCounter = 1;
+
+        private void rpcLogger(string log)
```

Comment:
This needs to be reworked. Do we have this run concurrently? If not we should just open the file and keep it open rather than reopening it per request and reading all potentially huge file each time!

Also can this be done using our logging framework with new logger to the file with some size restrictions and roll-over files?

Reply:
reworked

Reply:
let us have understanding for Mateusz's C#! well done, needed a few touches